src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,219 @@
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't create "
+                    + car + ". ", throwable);
+        }
+        addDriversToCar(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getQuery = "SELECT cars.id AS id, cars.model AS model, "
+                + "manufacturers.id AS manufacturer_id,\n"
+                + "manufacturers.name AS name, manufacturers.country AS country FROM cars \n"
+                + "JOIN manufacturers ON manufacturers.id = cars.manufacturer_id\n"
+                + "WHERE cars.id = ? AND cars.deleted = false;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getStatement = connection.prepareStatement(getQuery)) {
+            getStatement.setLong(1, id);
+            ResultSet resultSet = getStatement.executeQuery();
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get driver by id: " + id, throwable);
+        }
+        if (car != null) {
+            car.setDrivers(getAllDriversByCar(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT * FROM cars\n"
+                + "JOIN manufacturers\n"
+                + "ON manufacturers.id = cars.manufacturer_id\n"
+                + "WHERE cars.deleted = false;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get a list of cars from carsDB.",
+                    throwable);
+        }
+        checkAndSetDrivers(cars);
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars "
+                + "SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateDriverStatement =
+                        connection.prepareStatement(query)) {
+            updateDriverStatement.setString(1, car.getModel());
+            updateDriverStatement.setLong(2, car.getManufacturer().getId());
+            updateDriverStatement.setLong(3, car.getId());
+            if (updateDriverStatement.executeUpdate() > 0) {
+                deleteDriversFromCar(car.getId());
```

Comment:
this code is duplicated, we don't need to update drivers twice, plus you opened nested connection here